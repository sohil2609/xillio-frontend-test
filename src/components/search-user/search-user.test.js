import React from 'react';
import { shallow } from 'enzyme';
import SearchUser from './search-user.component';

const setUp = (props = {}) => {
	const component = shallow(<SearchUser {...props} />);
	return component;
};

describe('Search User Component', () => {
	let component;
	let mockProps;
	beforeEach(() => {
		mockProps = {
			query: 'xyz',
			onChange: jest.fn(),
			onSubmit: jest.fn()
		};
		component = setUp(mockProps);
	});
	it('Should test snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('Should render button without errors', () => {
		const wrapper = component.find('[data-test="searchButton"]');
		expect(wrapper.length).toBe(1);
	});

	it('Should render text input without errors', () => {
		const wrapper = component.find('[data-test="textInputComponent"]');
		expect(wrapper.length).toBe(1);
	});
});
