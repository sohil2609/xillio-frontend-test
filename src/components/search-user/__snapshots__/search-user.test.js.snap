// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search User Component Should test snapshot 1`] = `ShallowWrapper {}`;
